src/widgets/NestedStringListArrayWidget.tsx

```diff
@@ -0,0 +1,791 @@
+/* istanbul ignore file */
+/* see cypress/component/StringListArrayWidget.cy.tsx **/
+import React, { useRef } from 'react'
+import { flushSync, render } from 'react-dom'
+import { Trigger as AccordionTriggerPrimitive } from '@radix-ui/react-accordion'
+import { ChevronDownIcon } from '@radix-ui/react-icons'
+import { useEventListener } from 'usehooks-ts'
+import styled from 'styled-components'
+
+import 'core-js/actual/set/intersection.js'
+import 'core-js/actual/set/difference.js'
+
+import { AccordionSingle, Checkbox, Label, WidgetTooltip } from '../index'
+
+import {
+  ActionButton,
+  InputGroup,
+  InputsGrid,
+  LabelWrapper,
+  Fieldset,
+  Legend,
+  Widget,
+  WidgetActionsWrapper,
+  WidgetHeader,
+  WidgetTitle,
+  Error,
+  ReservedSpace
+} from './Widget'
+
+import {
+  isDisabled,
+  getPermittedBulkSelection,
+  isAllSelected,
+  useBypassRequired,
+  useWidgetSelection
+} from '../utils'
+import {
+  StringListWidgetConfiguration,
+  StringListWidgetDetails
+} from './StringListWidget'
+
+declare global {
+  interface Set<T> {
+    intersection(other: Set<T>): Set<T>
+    difference(other: Set<T>): Set<T>
+  }
+}
+
+export interface StringListArrayWidgetGroupDetails {
+  columns: number
+  label: string
+  labels: Record<string, string | undefined>
+  values: string[]
+}
+
+export interface StringListArrayWidgetDetails {
+  accordionGroups: boolean
+  accordionOptions: {
+    openGroups: string[]
+    searchable: boolean
+  }
+  displayaslist: boolean
+  groups:
+    | StringListArrayWidgetGroupDetails[]
+    | Pick<StringListArrayWidgetConfiguration, 'label' | 'details'>[]
+  id: number
+}
+
+export interface StringListArrayWidgetConfiguration {
+  type: 'StringListArrayWidget'
+  name: string
+  label: string
+  help?: string | null
+  required: boolean
+  details: StringListArrayWidgetDetails
+}
+
+interface StringListArrayWidgetProps {
+  configuration: StringListArrayWidgetConfiguration
+  /**
+   * Permitted selections for the widget.
+   */
+  constraints?: string[]
+  /**
+   * Whether the underlying fieldset should be functionally and visually disabled.
+   */
+  fieldsetDisabled?: boolean
+  /**
+   * Whether to hide the widget label from ARIA.
+   */
+  labelAriaHidden?: boolean
+  /**
+   * When true, bypass the required attribute if all options are made unavailable by constraints.
+   */
+  bypassRequiredForConstraints?: boolean
+
+  /**
+   * When true, shows the active selection count for closed accordions.
+   */
+  renderActiveSelectionsCount?: boolean
+}
+
+const getGroups = (
+  groups: StringListArrayWidgetConfiguration['details']['groups'],
+  result: StringListArrayWidgetGroupDetails[] = []
+) => {
+  for (const group of groups) {
+    if ('details' in group) {
+      getGroups(
+        group.details
+          .groups as StringListArrayWidgetConfiguration['details']['groups'],
+        result
+      )
+    } else {
```

Comment:
let's see if we can avoid `else` and `push`. Also, this function can be put under a unit test